kernel/src/vm/vmar/vm_mapping.rs

```diff
@@ -29,71 +23,59 @@ use crate::{
     },
 };
 
-/// A `VmMapping` represents mapping a range of physical pages into a `Vmar`.
+/// Mapping a range of physical pages into a `Vmar`.
 ///
-/// A `VmMapping` can bind with a `Vmo` which can provide physical pages for mapping.
-/// Otherwise, it must be an anonymous mapping and will map any empty physical page.
-/// A `VmMapping` binding with a `Vmo` is called VMO-backed mapping. Generally, a VMO-backed
-/// mapping is a file-backed mapping. Yet there are also some situations where specific pages
-/// that are not in a file need to be mapped. e.g:
-/// - Mappings to the VDSO data.
-/// - Shared anonymous mappings. because the mapped pages need to be retained and shared with
-///   other processes.
+/// A `VmMapping` can bind with a `Vmo` which can provide physical pages for
+/// mapping. Otherwise, it must be an anonymous mapping and will map any empty
+/// physical page. A `VmMapping` binding with a `Vmo` is called VMO-backed
+/// mapping. Generally, a VMO-backed mapping is a file-backed mapping. Yet
+/// there are also some situations where specific pages that are not in a file
+/// need to be mapped. e.g:
+///  - Mappings to the VDSO data.
+///  - Shared anonymous mappings. because the mapped pages need to be retained
+///    and shared with other processes.
 ///
 /// Such mappings will also be VMO-backed mappings.
+///
+/// This type controls the actual mapping in the [`VmSpace`]. It is a linear
+/// type and cannot be [`Drop`]. To remove a mapping, use [`Self::unmap`].
+#[derive(Debug)]
 pub(super) struct VmMapping {
-    inner: RwLock<VmMappingInner>,
-    /// The parent VMAR. The parent should always point to a valid VMAR.
-    parent: Weak<Vmar_>,
-    /// Specific physical pages that need to be mapped.
-    /// If this field is `None`, it means that the mapping is
-    /// an independent anonymous mapping.
+    /// The size of mapping, in bytes. The map size can even be larger than the
+    /// size of VMO. Those pages outside VMO range cannot be read or write.
+    ///
+    /// Zero sized mapping is not allowed. So this field is always non-zero.
+    map_size: usize,
```

Comment:
Since it's always zero, may using [NonZeroUsize](https://doc.rust-lang.org/std/num/type.NonZeroUsize.html#) is better than `usize`